Libraries/Image/RCTImageView.m

```diff
@@ -347,6 +347,14 @@ - (void)imageLoaderLoadedImage:(UIImage *)loadedImage error:(NSError *)error for
   }
 
   if (error) {
+    if (_pendingImageSource.request && _imageSource.request && ![_pendingImageSource.request.URL.absoluteString isEqual: _imageSource.request.URL.absoluteString]) {
```

Comment:
Could you explain me why this condition is necessary? I am having hard time understanding why it is needed.

Otherwise
You can get rid of `. absoluteString` as `NSURL` has `isEqual` defined (https://developer.apple.com/documentation/foundation/nsurl/1806686-isequal?language=objc).

Also I don't think we have to check whether `_pendingImageSource.request` and `_imageSource.request` are nils.

Reply:
I think the `Image` should really show whether the image is loaded successfully. If the loading fails and the URL is not equal to the previous one, then the previous image should not be displayed again, because the developer expects to display the new image, But the new image fails to load. . Usually developers can use `onError`'s callbacks to handle failures, but when the developer doesn't handle the error, the image displayed is actually wrong.

Also, thank you for your suggestion. I got a new skill  : )